community/modules/scripts/spack-install/variables.tf

```diff
@@ -178,11 +190,32 @@ EOT
 
 variable "environments" {
   description = "Defines a spack environment to configure."
```

Comment:
Given this has gotten more complicated, the description should include more information on expected keys and values.